### PR TITLE
Add dap-source command

### PR DESCRIPTION
**Pull request by forestaa**
This PR is related to https://github.com/phoityne/hdx4vsc/issues/27

Add the command `dap-source` which returns the content of the loaded modules to support [Source Request](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Source) by haskell-debug-adapter.
This PR depends https://github.com/phoityne/haskell-dap/pull/13

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -9,16 +9,10 @@ import Control.Concurrent
...
 
-import Paths_ghci_dap (version)
-import Data.Version
-
 -- |
 --  Main
---
 main :: IO ()
 main = do
-  putStrLn $ "[DAP][INFO] start ghci-dap-" ++ showVersion version ++ "."
```
Comment (phoityne) [inline app/Main.hs:20]: I think it's ok just showing it to the stderr handler. (hPutStrLn stderr)

**Review by phoityne — COMMENTED** [review]
Thank you for your additional commits.
Specifically for this issue, I'm also going to add the following changes.
If you don't mind, try this one.
```
$ git diff Command.hs
diff --git a/app/GHCi/DAP/Command.hs b/app/GHCi/DAP/Command.hs
index 87dc6ed..668f088 100644
--- a/app/GHCi/DAP/Command.hs
+++ b/app/GHCi/DAP/Command.hs
@@ -13,6 +13,7 @@ import qualified GHCi.UI.Monad as Gi hiding (runStmt)
 import Control.Concurrent
 import Control.Monad
 import System.Console.Haskeline
+import System.Directory
 import qualified Data.Map as M
 import qualified Data.List as L

@@ -457,12 +458,21 @@ dapStackTraceCmd_ _ = do
                      then hists
                      else resume2stackframe r : hists
       let traceWithId = setFrameIdx 0 traces
-
+      traceWithId2 <- liftIO $ mapM convertToAbs traceWithId
       return $ Right D.defaultStackTraceResponseBody {
-          D.stackFramesStackTraceResponseBody = traceWithId
-        , D.totalFramesStackTraceResponseBody = length traceWithId
+          D.stackFramesStackTraceResponseBody = traceWithId2
+        , D.totalFramesStackTraceResponseBody = length traceWithId2
         }

+    -- |
+    --
+    convertToAbs :: D.StackFrame -> IO D.StackFrame
+    convertToAbs sf@D.StackFrame{D.sourceStackFrame=ssf}
+      | D.pathSource ssf == "UnhelpfulSpan" = return sf
+      | otherwise = do
+          ps <- canonicalizePath $ D.pathSource ssf
+          return sf{D.sourceStackFrame=ssf{D.pathSource = ps}}
+
```

File: app/Main.hs (hunk 1 above)
Comment (forestaa) [reply app/Main.hs:20]: It makes sense. Thank you for your comment. https://github.com/phoityne/ghci-dap/pull/8/commits/301cedaa39d342c40202686d14c8e90a6829331c

**Comment by forestaa** [comment]
I add the following commit and it works well.
https://github.com/phoityne/ghci-dap/pull/8/commits/e8604de6225d9eead7ecfc97f9983301db35c170

**Comment by phoityne** [comment]
Thank you. merged.

**Pull request merged**